src/nvim/path.c

```diff
@@ -1520,31 +1520,26 @@ find_file_name_in_path (
   return file_name;
 }
 
-/*
- * Check if the "://" of a URL is at the pointer, return URL_SLASH.
- * Also check for ":\\", which MS Internet Explorer accepts, return
- * URL_BACKSLASH.
- */
-int path_is_url(char_u *p)
+// Check if the "://" of a URL is at the pointer, return URL_SLASH.
+// Also check for ":\\", which MS Internet Explorer accepts, return
+// URL_BACKSLASH.
+int path_is_url(char *p)
 {
-  if (STRNCMP(p, "://", (size_t)3) == 0)
+  if (strncmp(p, "://", (size_t)3) == 0)
     return URL_SLASH;
-  else if (STRNCMP(p, ":\\\\", (size_t)3) == 0)
+  else if (strncmp(p, ":\\\\", (size_t)3) == 0)
     return URL_BACKSLASH;
   return 0;
 }
 
-/*
- * Check if "fname" starts with "name://".  Return URL_SLASH if it does.
- * Return URL_BACKSLASH for "name:\\".
- * Return zero otherwise.
- */
-int path_with_url(char_u *fname)
+// Check if "fname" starts with "name://".  Return URL_SLASH if it does.
+// Return URL_BACKSLASH for "name:\\".
+// Return zero otherwise.
+int path_with_url(char *fname)
 {
-  char_u *p;
+  char *p;
 
-  for (p = fname; isalpha(*p); ++p)
-    ;
+  for (p = fname; isalpha(*p); p++) {}
```

Comment:
If I'm understanding you correctly I don't think that will work.  `p` gets used in the return statement after the for loop.  If I move it into the init, that will limit the scope to the loop, so the return will fail.